### DISCUSSION TITLE
vrnetlab version not seen within docker image

### DISCUSSION THREAD

**Discussion by flosch-dev** [hellt/vrnetlab#253] [Ideas]
Maybe there is already a way to identify which vrnetlab version was used to build a specific docker image. Currently the image only contains the kind and the release as docker image tag. But creating the same docker image with different vrnetlab version can result in different docker images. 
Is there a way to identify which vrnetlab version was used to create the docker image? Would it make sense to somehow include the information, for example copy a simple text file and output the content as debug to stdout?

I also willing to contribute to this change, if this is useful?

**Comment by hellt** [comment]
yes, this is definitely useful
I think it should be a commit hash that is included in the boot log and can be a file (`~/vrnetlab-version`)

**Comment by flosch-dev** [comment]
I created a change on my fork, but before creating the PR, I wanted to check if it goes in right direction.
Now it logs the following when launch.py is started:
```
2024-09-23 13:00:44,559: launch     DEBUG    vrnetlab VERSION: 'Commit: 26f9d10c7f5ce72dc6e19f679cf2def58439a8b8 from Sat, 21 Sep 2024 08:50:32 +0200'
```
The Change is here:
https://github.com/flosch-dev/vrnetlab/commit/6c7eee7a25c84a6caf2f68bf8031d83a88682fbc

Is commit hash + date enough?
I assume, it would need to be adapted for all other kinds manually?

**Reply by hellt** [reply]
I think the git version should be done in a common make file, not per-node one

Also the print of a version should happen from the common.py file, and not from a node specific one

**Reply by flosch-dev** [reply]
https://github.com/flosch-dev/vrnetlab/commit/732a7ab38498202ca98849f73a8120a1b74fc24b
for make file, I already used the makefile.include, which can be used by all kinds.
But still, the Dockerfile (copy the vrnetlab-version file into image) is kind specific. Also I added a func into vrnetlab, but to generate the output, launch.py needs adaption.

**Comment by flosch-dev** [comment]
How about adding a label "vrnetlab-version" to the docker container? Content can be the same as above (commit hash + commit date).
It could be done in makefile.include and would work for any kind.

```
            "Labels": {
                "org.opencontainers.image.authors": "roman@dodin.dev",
                "vrnetlab-version": "Commit: 26f9d10c7f5ce72dc6e19f679cf2def58439a8b8 from Sat, 21 Sep 2024 08:50:32 +0200"
            }
```

**Reply by hellt** [reply]
That's a good idea
